src/ui/AuthenticationProvider/AuthenticationProvider.ts

```diff
@@ -188,6 +197,19 @@ export class AuthenticationProvider extends Component {
     localStorage.setItem(accessTokenStorageKey, accessToken);
   }
 
+  private removeHandshakeTokenFromUrl() {
```

Comment:
Could this be replaced by a regex ? 

Not sure this works for all cases, please run the UT first. I assume the token is always alpha numerical.

```javascript
const adjustedHash = hash.replace(/\&?handshake_token=[a-z0-9]+\&?/, '')

[...]
```


Or do you find that more complex overall ?

Reply:
Hmm, a regex would be more elegant, but I'm not sure how to manage the case where the handshake token is in the middle of the hash:

```
#a=b&handshake_token=token&c=d
```
The regex would adjust the string to:
```
a=bc=d
```

We can assume the token is always at the end, but if the search api ever changes this, we might need to adjust the UI code